File: Greedy, BruteForce/Java/1783 병든 나이트.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(); int m = sc.nextInt();
        int answer = 0;
        
        
        if(n == 1) answer = 1;
        else if(n == 2) answer = Math.min(4, ((m-1)/2)+1);
        else {
            if(m <= 6) answer = Math.min(4, m);
            else answer = m - 2;
        }
        
        System.out.println(answer);
    }
}

File: _etc/Kotlin/1783 병든 나이트.kt
import java.util.*;
import kotlin.math.*;

fun main(args: Array<String>) {
    var sc = Scanner(System.`in`)
    var n = sc.nextInt(); var m = sc.nextInt()
    var answer: Int;
    if(n == 1) answer = 1;
    else if(n == 2) answer = min(4, ((m-1)/2)+1)
    else {
        if(m > 6) answer = m-2;
        else answer = min(4, m);
    }
    print(answer);
}
